api/types.proto

```diff
@@ -277,6 +277,41 @@ message UpdateConfig {
 
 	// Amount of time between updates.
 	Duration delay = 2 [(gogoproto.nullable) = false];
+
+	enum OnFailure {
+		PAUSE = 0;
+		CONTINUE = 1;
+		// TODO(aaronl): Add ROLLBACK as a supported failure mode.
+		// (#486)
+	}
+
+	// OnFailure is the action to take when an update failures. Currently,
+	// a failure is defined as a single updated task failing to reach the
+	// RUNNING state. In the future, there will be configuration to define
+	// what is treated as a failure (see #486 for a proposal).
+	OnFailure on_failure = 3;
+}
+
+// UpdateStatus is the status of an update in progress.
+message UpdateStatus {
+	enum UpdateState {
+		NEW = 0;
+		RUNNING = 1;
+		PAUSED = 2;
+		COMPLETED = 3;
+		// TODO(aaronl): add ROLLING_BACK, ROLLED_BACK as part of
+		// rollback support.
+	}
+
+	// State is the state of this update. It indicates whether the
+	// update is in progress, completed, or is paused.
+	UpdateState state = 1;
+
+	// Message explains how the update got into its current state. For
+	// example, if the update is paused, it will explain what is preventing
+	// the update from proceeding (typically the failure of a task to start up
+	// when OnFailure is PAUSE).
+	string message = 2;
```

Comment:
Started and last updated would be different timestamps, but I really don't want to add "last updated" right now. The reason is that updating the service is a relatively heavyweight operation, since it triggers reconciliation and in fact restarts the rolling update (unless it's paused). So until we have a better way to work around this, we should avoid updating the service too frequently.